datatree/tests/test_datatree.py

```diff
@@ -480,8 +480,24 @@ def test_methods(self):
         assert ds.mean().identical(dt.ds.mean())
         assert type(dt.ds.mean()) == xr.Dataset
 
+    def test_getitem(self):
+        dt = create_test_datatree()
+        da = dt.ds["a"]
+        print(da)
+        expected = xr.DataArray(dims=("y",), data=[6, 7, 8])
+        print(expected)
+        xrt.assert_equal(da, expected)
+
+    # @pytest.mark.xfail
     def test_get_items_in_other_nodes(self):
-        ...
+        dt = create_test_datatree()
+        node_dsv = dt["set1"].ds
+
+        xrt.assert_identical(node_dsv["../a"], dt["a"])
+
+        # don't allow retrieving other DataTree nodes
+        with pytest.raises(KeyError):
+            node_dsv["../set2"]
```

Comment:
Calling `.ds.__getitem__` anywhere in these tests currently raises a `RecursionError`, and the cause is something complicated to do with `xarray.Dataset`'s `__getattr__` manipulation interacting with me adding a new `_wrapping_node` attribute to my `DatasetView` subclass.